### DISCUSSION TITLE
Integration läuft nicht

### DISCUSSION THREAD

**Discussion by Volkusch** [pail23/stiebel_eltron_isg_component#359] [General]
Hallo,

ich bekommen immer nachfolgende Fehlermeldung, was läuft hier falsch?

![grafik](https://github.com/user-attachments/assets/d8064c5d-81cf-4b53-9310-492c61aff570)
 
Das ISG ist an Stiebel Eltron WPF 5E Bj. 2013 angeschlossen, eine EMI Erweiterung ist nicht installiert, weil die Wärmepumpe lt. Stiebel Eltron zu alt ist.

**Comment by mithri** [comment]
Richtige IP Adresse?

**Comment by Volkusch** [comment]
> Richtige IP Adresse?

Ja, das ist die IP über die ich die WP im Browser aufrufe.
![grafik](https://github.com/user-attachments/assets/428ee2c3-7d07-45b0-abe3-bd8c4df38b84)

**Comment by mithri** [comment]
Steht im Debug Protokoll mehr? Einmal aktivieren und das Plugin neu laden und dann das Debug Protokoll runterladen und reinschauen bzw. hier posten.

**Comment by Volkusch** [comment]
Ich bin Home Assistant Neuling...
Meist du das?
![grafik](https://github.com/user-attachments/assets/99f7bbc4-2bdf-40f8-ba35-5073675892f8)

**Comment by Volkusch** [comment]
Oder das?
2025-01-06 19:50:13.814 WARNING (MainThread) [pymodbus.logging] Failed to connect [Errno 111] Connect call failed ('192.168.178.20', 502)
2025-01-06 19:50:19.041 WARNING (MainThread) [pymodbus.logging] Failed to connect [Errno 111] Connect call failed ('192.168.178.20', 502)
2025-01-06 19:50:29.512 WARNING (MainThread) [pymodbus.logging] Failed to connect [Errno 111] Connect call failed ('192.168.178.20', 502)
2025-01-06 19:50:49.881 WARNING (MainThread) [pymodbus.logging] Failed to connect [Errno 111] Connect call failed ('192.168.178.20', 502)
2025-01-06 19:51:30.366 WARNING (MainThread) [pymodbus.logging] Failed to connect [Errno 111] Connect call failed ('192.168.178.20', 502)
2025-01-06 19:52:50.601 WARNING (MainThread) [pymodbus.logging] Failed to connect [Errno 111] Connect call failed ('192.168.178.20', 502)
2025-01-06 19:54:10.913 WARNING (MainThread) [pymodbus.logging] Failed to connect [Errno 111] Connect call failed ('192.168.178.20', 502)
2025-01-06 19:55:30.453 ERROR (MainThread) [pysmaplus.device_ennexos] Request to https://192.168.178.66/api/v1/measurements/live did not return a valid json. Code 401
2025-01-06 19:55:31.265 WARNING (MainThread) [pymodbus.logging] Failed to connect [Errno 111] Connect call failed ('192.168.178.20', 502)
2025-01-06 19:55:35.553 ERROR (MainThread) [pysmaplus.device_ennexos] Request to https://192.168.178.66/api/v1/measurements/live did not return a valid json. Code 401
2025-01-06 19:56:51.700 WARNING (MainThread) [pymodbus.logging] Failed to connect [Errno 111] Connect call failed ('192.168.178.20', 502)
2025-01-06 19:58:11.960 WARNING (MainThread) [pymodbus.logging] Failed to connect [Errno 111] Connect call failed ('192.168.178.20', 502)
2025-01-06 19:59:32.353 WARNING (MainThread) [pymodbus.logging] Failed to connect [Errno 111] Connect call failed ('192.168.178.20', 502)
2025-01-06 20:00:52.794 WARNING (MainThread) [pymodbus.logging] Failed to connect [Errno 111] Connect call failed ('192.168.178.20', 502)
2025-01-06 20:02:12.877 WARNING (MainThread) [pymodbus.logging] Failed to connect [Errno 111] Connect call failed ('192.168.178.20', 502)

**Comment by mithri** [comment]
Da kommt ein 401 zurück. Hast du Benutzername und Passwort auf dem ISG eingerichtet? Wenn ja, bitte mal entfernen und schauen ob es dann geht.
Der Log ist übrigens der richtige, danke.

**Comment by Volkusch** [comment]
Nein, ich habe da keinen Benutzername mit Passwort vergeben.
![grafik](https://github.com/user-attachments/assets/7550a01a-4c51-42c9-b19a-921a2adaa545)

**Comment by Volkusch** [comment]
Die 401 steht doch aber mit einem anden Gerät in Verbindung (IP 192.168.178.66), das ist meine SMA Wallbox...
Gibt es da möglicherweise einen Konflikt swischen dem ISG und der Wallbox?

**Comment by mithri** [comment]
Ok, falsch geschaut. Sorry. Dein HA kann einfach keine Verbindung zum ISG aufbauen, hast du die Verbindung zwischen HA und dem ISG geprüft?

**Comment by Volkusch** [comment]
Wie genau kann ich die Verbindung explizit zwischen HA und dem ISG prüfen?
HA, ISG und alle andern bisher in HA eingebundenen Geräte sind alle über eine FritzBox in einem Netzwerk verbunden, das ISG bzw. seine IP-Adresse ist über den Browser und CMG (PING) erreichbar.

**Comment by Volkusch** [comment]
Ich habe das mal hiermit ausprobiert, damit wird die IP (ISG) gefunden.
![grafik](https://github.com/user-attachments/assets/ac687572-9418-4417-95be-bc83f904722a)

**Comment by mithri** [comment]
Ok, das geht. Du könntest auf dem gleichen Wege überprüfen ob der Port 502 offen und erreichbar ist:

binary_sensor:
  - platform: tcp
    host: 192.168.178.xxx
    port: 502
    name: Port 502 Verfügbar
    payload: ""

Wenn das auch geht, dann weiß ich leider nicht warum er keine Verbindung aufbaut. Welche HA Version hast du? Ich habe noch 2024.12.3, weil in neueren Versionen sich die Modbus Bibliothek geändert hat, vlt. ist die Integration noch nicht kompatibel mit neueren Versionen.

**Comment by Volkusch** [comment]
Die Anwendung "Ping (ICMP)" findet zwar die IP mit dem Port 192.168.178.20:502 aber die Verbindung ist dauerhaft getrennt!?
![grafik](https://github.com/user-attachments/assets/f58d51ae-5304-4346-b866-6e913f74e7d9)

**Comment by Volkusch** [comment]
Meine HA Version:
![grafik](https://github.com/user-attachments/assets/3fec4eb7-3172-42fa-983a-ddbb77c47705)

**Comment by Karssi** [comment]
Also ich habe die gleiche HA version wie Volkusch und habe leider auch das gleiche Problem. :-(

**Comment by Volkusch** [comment]
Kurzes Update…
In anderen Foren hatte ich gelesen, dass es möglicherweise auch an der Softwareversion des ISG liegen könnte… Gesagt getan… Habe mein ISG von Stiebel updaten lassen (von v.7.0.0 auf v.12.2.3.), auch mit der neuen Version die gleiche Fehlermeldung…😕
Vielleicht hat ja noch jemand einen Lösungsvorschlag?

**Comment by guelz** [comment]
Home Assistant 2025.1.x kommt mit einer inkompatiblen pymodbus Version! Im Moment hilft nur 2024.12.5 wiederherzustellen und dort zu bleiben. Vorläufig..
Gleiches Problem bei anderen Integrationen [huawei_solar](https://github.com/wlcrs/huawei_solar/issues/873)

**Comment by mithri** [comment]
Danke @guelz für die Bestätigung, ich sehe das genauso und hatte das auch schon vor 2 Wochen hier geschrieben: https://github.com/pail23/stiebel_eltron_isg_component/issues/343#issuecomment-2580851993

Bitte macht doch ein Downgrade euer HA Instanz und testet es dann nochmal.

@pail23 Is a compatibility fix für newer pymodbus already underway? Or should someone else take a look?

**Comment by Volkusch** [comment]
Ah, danke für die Info... Dann warte ich mal ab.

**Comment by mrebbert** [comment]
Hey,

mit "ping" testest du eher die generelle Erreichbarkeit auf IP-Ebene, aber nicht die Verbindung zu einem TCP-Port.

Kannst Du nochmal testen, ob du den Port auf dem ISG erreichst?
bspw. mit 
`nc -v 192.168.178.20 502`
vom Server, auf dem HA läuft oder wie [oben](https://github.com/pail23/stiebel_eltron_isg_component/issues/343#issuecomment-2580851993) beschrieben direkt aus HA.

Ist modbus in der "Servicewelt" aktiviert? (Einstellungen->Energiemanagement)

**Comment by MarkoThaZ** [comment]
Hi 

Klingt einwenig banal, aber hatte auch den Fehler aber erst seit dem neustem Update vom HA und Steibel HACS.

Konnte es über CMD befehl tenenet mit dem port nicht erreichen.

IP uüber chrome ohne probleme.

ISG neugestartet und mich mit dem Steibel APP angemeldet und auf einmal wie zauber hand funktioniert es wieder

**Comment by Volkusch** [comment]
> Hey,
> 
> mit "ping" testest du eher die generelle Erreichbarkeit auf IP-Ebene, aber nicht die Verbindung zu einem TCP-Port.
> 
> Kannst Du nochmal testen, ob du den Port auf dem ISG erreichst? bspw. mit `nc -v 192.168.178.20 502` vom Server, auf dem HA läuft oder wie [oben](https://github.com/pail23/stiebel_eltron_isg_component/issues/343#issuecomment-2580851993) beschrieben direkt aus HA.
> 
> Ist modbus in der "Servicewelt" aktiviert? (Einstellungen->Energiemanagement)

Hi,

ich muss vorwegschicken, ich bin blutiger HA Anfänger… Leider ist mir nicht ganz klar, wie ich den Port vom Server testen soll, wo und wie soll ich den folgenden Code eingeben, bzw. testen?

platform: tcp
host: 192.168.178.xxx
port: 502
name: Port 502 Verfügbar
payload: ""

Ich hatte bereits die o. g. Zeilen kopiert und in meine „configuration.yaml“ eingefügt, bekomme dann aber immer eine Fehlermeldung, dass da etwas falsch ist...  Wie gesagt, ich bin was HA und YAML-Eingaben betrifft ein absoluter Anfänger…

In Servicewelt im Bereich Einstellungen gibt es bei mir den Punkt „Energiemanagement“ gar nicht, vielleicht ist meine WP zu alt!?

![Image](https://github.com/user-attachments/assets/45dce8f8-e94f-48ca-83e3-9dd8762bf241)

**Comment by mrebbert** [comment]
Zum Port Test:

```
command_line:
  - binary_sensor:
      name: is_modbus_available
      command: 'nc -z 192.168.178.xxx 502 > /dev/null 2>&1 && echo ON || echo OFF'
      scan_interval: 30
```

Wenn 'nc' auf Deinem Host installiert ist (wahrscheinlich), solltest Du einen binary_sensor.is_modbus_available nach restart finden.

Hinsichtlich Servicewelt bin ich mir nicht sicher: Du müsstest irgendwo Einstellungen zu SG ready finden.
Ansonsten hast Du ggf. keine SG Ready-Schnittstelle.

<img width="968" alt="Image" src="https://github.com/user-attachments/assets/39ca0de5-8cb5-41cb-ac0c-d36107bb0e99" />

**Comment by Volkusch** [comment]
Danke für deine Mühe,

weder über HA noch direkt aus Windnows (Telnet-Client) heraus ist der Port 502 zu erreichen😩

Ich hatte auch noch einmal Kontakt mit Stiebel Eltron, meine WP aus 2013 war nicht SG-Ready und kann auch nicht geupdatet werden, sie ist schlecht weg "zu alt"... Also auch nicht mit ModBus☹

Ich wollte ja auch nichts schalten, es hätte mir ja völlig ausgereicht die anderen Daten (z. B. Vor- u. Rücklauftemperaturen) mitzuloggen...
Kennt evt. Jeman eine Möglichkeit diese Daten aus der WP in HA zubekommen?

**Comment by reneschuster** [comment]
Direkt über den CAN Bus, aber scheint nicht so easy zu sein.

https://community.home-assistant.io/t/configured-my-esphome-with-mcp2515-can-bus-for-stiebel-eltron-heating-pump/366053

https://www.youtube.com/watch?v=81KwHOngOlU

**Comment by Volkusch** [comment]
> Direkt über den CAN Bus, aber scheint nicht so easy zu sein.
> 
> https://community.home-assistant.io/t/configured-my-esphome-with-mcp2515-can-bus-for-stiebel-eltron-heating-pump/366053
> 
> https://www.youtube.com/watch?v=81KwHOngOlU

Danke für den Tip,

auf diese Möglichkeit bin ich am Anfang auch schon gestoßen, habe es dann aber eben genau aus den von dir genannten Gründen verworfen… Ich dachte dann aber mit dem ISG und der o. g. Integration funktioniert es aber leider war das nicht so…
Wenn ich mal irgendwann mehr Erfahrung mit HA gesammelt und viel Zeit haben sollte, versuche ich es vielleicht einmal auf diesem Weg!?

**Comment by Volkusch** [comment]
Fall aber jemand eine Idee hat aus einer etwas „ältere“ (2012) Stiebel Eltron WP Daten zu bekommen (Modbus fällt ja nun leider aus) dann her damit, ein ISG ist vorhanden und damit sehe ich ja auch schon sehr viele Datenpunkten nur leider eben nicht in HA…😥

**Comment by mithri** [comment]
Ich sehe dann nur die Möglichkeit per Canbus an die Daten zu kommen, eben wie es das ISG auch selber macht. Echt seltsam warum das ISG die Daten nicht per Modbus für deine WPF 5 rausgibt. Steht aber so auch in der Kompatibilitätsliste von Stiebel.

https://www.stiebel-eltron.de/de/home/service/smart-home/kompatibilitaetslisten.html

**Comment by reneschuster** [comment]
Da wäre dann noch die [Scrape ](https://www.home-assistant.io/integrations/scrape/)Integration um die Daten von der ISG HTML-Webseite zu extrahieren. 
Hier ein [Beispiel ](https://github.com/pail23/stiebel_eltron_isg_component/issues/332#issuecomment-2522606995)um die folgenden Werte auszulesen:

WM HEIZEN TAG
WM WW TAG
P HEIZUNG TAG
P WW TAG

**Comment by thitcher** [comment]
> Hi
> 
> Klingt einwenig banal, aber hatte auch den Fehler aber erst seit dem neustem Update vom HA und Steibel HACS.
> 
> Konnte es über CMD befehl tenenet mit dem port nicht erreichen.
> 
> IP uüber chrome ohne probleme.
> 
> ISG neugestartet und mich mit dem Steibel APP angemeldet und auf einmal wie zauber hand funktioniert es wieder

Thats it!
Kollege hatte selbes Problem. Integration gelöscht, ISG neugestartet, Inegration wieder eingerichtet --> läuft

**Comment by Optic00** [comment]
Closing this as resolved. The thread established that TCP port 502 is not available on this installation, and the reporter was told by Stiebel that the 2013 WPF 5 cannot be upgraded for Modbus. This Modbus integration therefore cannot reach it. CAN bus or reading the ISG web interface remain separate alternatives. Thanks to everyone who helped narrow it down.
